### PR TITLE
Most updated with the navigation bar 

### PR DESCRIPTION
**Pull request by ysilksa**
Has a navigation bar, the working version is the hw 4 branch but it is much less updated than this branch. 

It stopped loading in my browser when I ran npm start, so I couldn't continue because I wasn't sure how the website was being updated.

### CODE REVIEW

**Comment by aanandp123** [comment]
Checking your browsers developer console might help you identify any errors that have been logged....Open your app (localhost) on google chrome->right click then inspect->click on the console tab on the window that pops up on the right. Sometimes the build can succeed but still have runtime errors that prevent app from loading

File: my-app/src/App.js [hunk 1]
```diff
@@ -1,25 +1,47 @@
...
+import Profile from './pages/Profile';
+import Discover from './pages/Discover';
+import AboutUs from './pages/AboutUs';
+import Home from './pages/Home';
+import "./NavBarStyles.css"
+import { Route, Routes } from "react-router-dom";
 
 function App() {
+  // to create light and dark mode 
+  // const [mode, setMode] = useState('light');
```
Comment (aanandp123) [inline my-app/src/App.js:13]: great use of dark and light mode!

File: my-app/src/App.js [hunk 2]
```diff
@@ -1,25 +1,47 @@
...
+
+  // const switchMode = () => {
+  //   if (mode === 'light') {
+  //     setMode('dark');
+  //   } else if (mode === 'dark') {
+  //     setMode('light');
+  //   }
+  // }
+
+  // useEffect(() => {
```
Comment (aanandp123) [inline my-app/src/App.js:23]: appreciate the `useEffect`!

File: my-app/src/App.js [hunk 3]
```diff
@@ -1,25 +1,47 @@
...
+    <>
+    <NavBar />
+    <div className = "container">
+      {/* style={{ backgroundColor: backgroundColor, color: textColor }}> */}
+      {/* <button onClick={switchMode}>Dark mode</button>  */}
+      <Routes>
+        <Route path = "/" element = {<Home />} />
+        <Route path = "/discover" element = {<Discover />} />
+        <Route path = "/profile" element = {<Profile />} />
+        <Route path = "/aboutus" element = {<AboutUs />} />
```
Comment (aanandp123) [inline my-app/src/App.js:40]: The Home, AboutUs and Discover components seem static and very basic. Would love to see you populate those pages with some information and important functionality!

File: my-app/src/pages/AboutUs.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import '../App.css';
+import './AboutUs.css'
+
+export default function AboutUs() {
+  // to create light and dark mode 
+  const [mode, setMode] = useState('light');
+
+  const switchMode = () => {
```
Comment (aanandp123) [inline my-app/src/pages/AboutUs.js:10]: you can shorten this function and make it more succinct with a simple ternary operator

File: my-app/src/App.js (hunk 3 above)
Comment (aanandp123) [reply my-app/src/App.js:40]: However, the file organization and component structure is great and well organized. Great job on getting the css styles done as well

**Comment by amyzliao** [comment]
You are free to merge this branch into main. We recommend you do so immediately, before you make any changes to your codebase for this assignment or future assignments.

If you have NOT started on assignment 5, the following steps will help you avoid tricky merge conflicts:
- If you have made local changes to your `hw-4` branch that are not published to github, first commit those changes and publish them to github. That should update this PR with your latest changes.
1. Merge this PR into main
2. Switch to the main branch: `git checkout main`
3. Download the latest main which should now have all your `hw-4` commits: `git pull`
4. Delete your local `hw-4` branch: `git branch -d hw-4`

If you HAVE started on assignment 5, the following steps will help you avoid tricky merge conflicts:
- **DO NOT MAKE ANY CHANGES TO YOUR `hw-4` BRANCH**. If you have already made changes to `hw-4`, you will have to resolve merge conflicts at step 6.
1. Merge this PR into main
2. Commit all your current changes for assignment 5 to the `hw-5` branch. Afterwards, make sure when you run `git status`, it says "nothing to commit, working tree clean".
3. Switch back to the main branch: `git checkout main`
4. Download the latest main which should now have all your `hw-4` commits: `git pull`
5. Switch back to your `hw-5` branch: `git checkout hw-5`
6. Rebase off of main: `git rebase main`
    - This is the trickiest bit. Ideally there should not be any conflicts.
    - If you have conflicts, the terminal output will tell you exactly which files conflict. Go to those files and resolve the conflicts. When you're done, run `git add . && git rebase --continue`
    - Either the rebase will finish or it will give you more merge conflicts to resolve. If it gives you more conflicts to resolve, repeat the above bullet point. It might make you resolve the same conflict more than once; that is normal behavior.
    - After you are fully finished with the rebase, run your code again and verify that it has the expected behavior. Bugs may have emerged during the rebase.
7. Force push your changes to the `hw-5` branch to github:
    - If this branch does not yet exist on github: `git push -u origin hw-5`
    - If this branch already exists on github: `git push --force-with-lease`

If you run into any issues, DM amy, ethan, or aanand on discord.

**Pull request merged**